modules/runner/src/integrationTest/java/org/apache/ignite/internal/ItTruncateRaftLogAndRestartNodesTest.java

```diff
@@ -0,0 +1,494 @@
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.raft.jraft.conf.ConfigurationManager;
+import org.apache.ignite.raft.jraft.core.NodeImpl;
+import org.apache.ignite.raft.jraft.option.LogStorageOptions;
+import org.apache.ignite.raft.jraft.option.NodeOptions;
+import org.apache.ignite.raft.jraft.option.RaftOptions;
+import org.apache.ignite.raft.jraft.storage.LogStorage;
+import org.apache.ignite.tx.TransactionOptions;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Class for testing various scenarios with raft log truncation and node restarts that emulate the situation with fsync disabled for raft
+ * groups associated with tables.
+ */
+public class ItTruncateRaftLogAndRestartNodesTest extends ClusterPerTestIntegrationTest {
+    private static final IgniteLogger LOG = Loggers.forClass(ItTruncateRaftLogAndRestartNodesTest.class);
+
+    private static final String ZONE_NAME = "TEST_ZONE";
+
+    private static final String TABLE_NAME = "TEST_TABLE";
+
+    @Override
+    protected int initialNodes() {
+        return 0;
+    }
+
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-24802")
+    @Test
+    void enterNodeWithIndexGreaterThanCurrentMajority() throws Exception {
+        cluster.startAndInit(3);
+
+        createZoneAndTablePerson(ZONE_NAME, TABLE_NAME, 3, 1);
+
+        cluster.transferLeadershipTo(2, cluster.solePartitionId());
+
+        var closableResources = new ArrayList<ManuallyCloseable>();
+
+        try {
+            TestLogStorageFactory testLogStorageFactoryNode0 = createTestLogStorageFactory(0, cluster.solePartitionId());
+            closableResources.add(testLogStorageFactoryNode0);
+
+            TestLogStorageFactory testLogStorageFactoryNode1 = createTestLogStorageFactory(1, cluster.solePartitionId());
+            closableResources.add(testLogStorageFactoryNode1);
+
+            long lastLogIndexBeforeInsertNode0 = raftNodeImpl(0, cluster.solePartitionId()).lastLogIndex();
+            long lastLogIndexBeforeInsertNode1 = raftNodeImpl(1, cluster.solePartitionId()).lastLogIndex();
+
+            Person[] people = generatePeople(10);
+            insertPeople(TABLE_NAME, people);
+
+            // Let's flush the state machine so that the raft log on a node cannot be truncated.
+            flushMvPartitionStorage(2, TABLE_NAME, 0);
+
+            stopNodes(0, 1, 2);
+
+            LogStorage logStorageNode0 = testLogStorageFactoryNode0.createLogStorage();
+            closableResources.add(logStorageNode0::shutdown);
+
+            LogStorage logStorageNode1 = testLogStorageFactoryNode1.createLogStorage();
+            closableResources.add(logStorageNode1::shutdown);
+
+            truncateRaftLogSuffixHalfOfChanges(logStorageNode0, lastLogIndexBeforeInsertNode0);
+            truncateRaftLogSuffixHalfOfChanges(logStorageNode1, lastLogIndexBeforeInsertNode1);
+
+            closeAllReversed(closableResources);
```

Comment:
Please add `closableResources.clear()` after this line, to make the code clearer